test/test_rpc_endpoints.py

```diff
@@ -0,0 +1,641 @@
+"""
+Tests RPC endpoints.
+"""
+from __future__ import annotations
+
+import json
+
+import pytest
+from starkware.starknet.public.abi import get_storage_var_address, get_selector_from_name
+
+from starknet_devnet.server import app
+
+from .util import (
+    load_file_content,
+)
+from .test_endpoints import send_transaction
+
+
+DEPLOY_CONTENT = load_file_content("deploy.json")
+INVOKE_CONTENT = load_file_content("invoke.json")
+
+
+def rpc_call(method: str, params: dict | list) -> dict:
+    """
+    Make a call to the RPC endpoint
+    """
+    req = {
+        "jsonrpc": "2.0",
+        "method": method,
+        "params": params,
+        "id": 0
+    }
+
+    resp = app.test_client().post(
+        "/rpc",
+        content_type="application/json",
+        data=json.dumps(req)
+    )
+    result = json.loads(resp.data.decode("utf-8"))
+    return result
+
+
+def gateway_call(method: str, **kwargs):
+    """
+    Make a call to the gateway
+    """
+    resp = app.test_client().get(
+        f"/feeder_gateway/{method}?{''.join(f'{key}={value}&' for key, value in kwargs.items()).rstrip('&')}"
+    )
+    return json.loads(resp.data.decode("utf-8"))
+
+
+@pytest.fixture(name="deploy_info")
+def fixture_deploy_info() -> dict:
+    """
+    Deploy a contract on devnet and return deployment info dict
+    """
+    resp = send_transaction(json.loads(DEPLOY_CONTENT))
+    deploy_info = json.loads(resp.data.decode("utf-8"))
+    return deploy_info
+
+
+@pytest.fixture(name="invoke_info")
+def fixture_invoke_info() -> dict:
+    """
+    Make a invoke transaction on devnet and return invoke info dict
+    """
+    invoke_tx = json.loads(INVOKE_CONTENT)
+    invoke_tx["calldata"] = ["0"]
+    resp = send_transaction(invoke_tx)
+    invoke_info = json.loads(resp.data.decode("utf-8"))
+    return invoke_info
+
+
+def pad_zero(felt: str) -> str:
+    """
+    Convert felt with format `0xValue` to format `0x0Value`
```

Comment:
Is always adding a leading zero the right solution? Shouldn't it not add the zero only if the length is not as required? I.e., it depends on how big the number is. (That's why sometimes converting the hex to int is used for testing).

Reply:
That one I'm unsure about. In theory, the RPC has [it's own definition of felt](https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L819) so ideally we should always be using `0x0` padded strings. 

But I don't have a strong opinion on that, I wrote this function because for some reason I was getting `0x0` strings from rpc (which I wanted), and `0x` from gateway, and for testing sake they needed to be uniform format.

Reply:
Alright, it can stay like this, doesn't look wrong.